### PR TITLE
Automation for CEPH-83594338

### PR DESCRIPTION
**Pull request by aarsharm**
Automation for Test Case CEPH-83594338 : Delete a Cloned Group Snapshot

Pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-OY1CKV

### CODE REVIEW

File: suites/squid/rbd/tier-2_rbd_group.yaml [hunk 1]
```diff
@@ -106,13 +106,12 @@ tests:
   - test:
       desc: Cloning an image from a Group Snapshot
       module: test_rbd_groups_image_clone.py
-      name: RBD group image clone create
-      polarion-id: CEPH-83594298
+      name: RBD group image clone create and delete
+      polarion-id: CEPH-83594298 & CEPH-83594338
```
Comment (sunilangadi2) [inline suites/squid/rbd/tier-2_rbd_group.yaml:110]: we won't be adding multiple ids in polarion-id
if you want to combine this test here please cover those delete test steps under this test CEPH-83594298 and keep only this test.

Comment (sunilangadi2) [reply suites/squid/rbd/tier-2_rbd_group.yaml:110]: as it fails here 
https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83594298%20&%20CEPH-83594338=

File: tests/rbd/test_rbd_groups_image_clone.py [hunk 2]
```diff
@@ -53,10 +53,23 @@ def test_rbd_groups_image_clone(rbd_obj, client, **kw):
         E.g: rbd info pool1/i1clone1
     12) Map the cloned images as a new block disk using rbd map <pool1/clone1>
     13) Mount directory onto that disk and create some files on that directory and write data to the files
+
+    Polarian CEPH-83594338 - Delete a Cloned Group Snapshot
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_groups_image_clone.py:57]: update here as well

File: tests/rbd/test_rbd_groups_image_clone.py [hunk 3]
```diff
@@ -194,19 +216,48 @@ def test_rbd_groups_image_clone(rbd_obj, client, **kw):
...
+                # Attempt to delete the cloned image.
+                log.info(f"Delete clone image: {clone}")
+                rbd.rm(pool=pool, image=clone)
+                # Verify that the cloned image has been removed and is no longer listed. E.g: rbd ls testpool
+                log.info(f"Verify cloned image {clone} has been removed")
+                out, err = rbd.ls(pool=pool, format="json")
+                out_json = json.loads(out)
+                log.info(out_json)
+                if clone in out_json:
+                    log.error(f"clone image {clone} still exist even after deleting")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_groups_image_clone.py:260]: please add `log.info` for the successfull deletion.

File: tests/rbd/test_rbd_groups_image_clone.py [hunk 4]
```diff
@@ -194,19 +216,48 @@ def test_rbd_groups_image_clone(rbd_obj, client, **kw):
...
+                # Remove the group snapshot snap_name from the group.
+                log.info(f"Remove group snapshot {group_snap}")
+                rbd.group.snap.rm(pool=pool, snap=group_snap)
+                # Verify that the snapshot has been moved to the trash.
+                log.info(f"Verify that the group snapshot {group_snap} is removed")
+                out, err = rbd.snap.ls(pool=pool, image=image, all="", format="json")
+                snap_json = json.loads(out)
+                for snap in snap_json:
+                    if snap["name"] == group_snap:
+                        log.error(
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_groups_image_clone.py:231]: please add `log.info` for successful group snap deletion.

File: tests/rbd/test_rbd_groups_image_clone.py [hunk 5]
```diff
@@ -194,19 +216,48 @@ def test_rbd_groups_image_clone(rbd_obj, client, **kw):
...
+                    return 1
+
+                # Unmap the cloned image.
+                log.info(f"Unmap clone image: {clone}")
+                rbd.unmap(pool=pool, image=clone)
+
+                # Attempt to delete the cloned image.
+                log.info(f"Delete clone image: {clone}")
+                rbd.rm(pool=pool, image=clone)
+                # Verify that the cloned image has been removed and is no longer listed. E.g: rbd ls testpool
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_groups_image_clone.py:254]: Example in the comment can be removed as pool name does not match

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

File: suites/squid/rbd/tier-2_rbd_group.yaml (hunk 1 above)
Comment (aarsharm) [reply suites/squid/rbd/tier-2_rbd_group.yaml:110]: Done

File: tests/rbd/test_rbd_groups_image_clone.py (hunk 2 above)
Comment (aarsharm) [reply tests/rbd/test_rbd_groups_image_clone.py:57]: Done

File: tests/rbd/test_rbd_groups_image_clone.py (hunk 3 above)
Comment (aarsharm) [reply tests/rbd/test_rbd_groups_image_clone.py:260]: Done

**Comment by aarsharm** [comment]
Pass logs after addressing comments:  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ISGYT8

**Review by sunilangadi2 — APPROVED** [review]
changes LGTM.

**Pull request merged**